client/i18n/schema-validation.ts

```diff
@@ -0,0 +1,264 @@
+import path from 'path';
+import { readFile } from 'fs/promises';
+import { availableLangs } from '../../config/i18n/all-langs';
+import introSchema from './locales/english/intro.json';
+import linksSchema from './locales/english/links.json';
+import metaTagsSchema from './locales/english/meta-tags.json';
+import motivationSchema from './locales/english/motivation.json';
+import translationsSchema from './locales/english/translations.json';
+import trendingSchema from './locales/english/trending.json';
+
+type MotivationalQuotes = { quote: string; author: string }[];
+
+/**
+ * Flattens a nested object structure into a single
+ * object with property chains as keys.
+ * @param {Object} obj Object to flatten
+ * @param {String} namespace Used for property chaining
+ */
+const flattenAnObject = (obj: Record<string, unknown>, namespace = '') => {
+  const flattened: Record<string, unknown> = {};
+  Object.keys(obj).forEach(key => {
+    const value = obj[key];
+    if (Array.isArray(value)) {
+      flattened[namespace ? `${namespace}.${key}` : key] = value;
+    } else if (typeof value === 'object') {
+      Object.assign(
+        flattened,
+        flattenAnObject(
+          value as Record<string, unknown>,
+          namespace ? `${namespace}.${key}` : key
+        )
+      );
+    } else {
+      flattened[namespace ? `${namespace}.${key}` : key] = value;
```

Comment:
```suggestion
    const value = obj[key];
    const field = namespace ? `${namespace}.${key}` : key
    if (Array.isArray(value)) {
      flattened[field] = value;
    } else if (typeof value === 'object') {
      Object.assign(
        flattened,
        flattenAnObject(value as Record<string, unknown>, field)
      );
    } else {
      flattened[field] = value;
```

Reply:
Thanks for the suggestion😊, I have changed it to this, and linter is also satisfied.